From 1a10e54389e09d8275788674b85464d978801da8 Mon Sep 17 00:00:00 2001
From: Mike Lothian <mike@fireburn.co.uk>
Date: Mon, 20 May 2024 10:57:20 +0100
Subject: [PATCH] Include cstdarg in precompiled.h
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This fixes the following build error:

src/Common/unix/FileStream_unix.cpp: In member function ‘void FileStream::writeStringFmt(const char*, ...)’:
src/Common/unix/FileStream_unix.cpp:184:9: error: ‘va_start’ was not declared in this scope
  184 |         va_start(args, format);
      |         ^~~~~~~~

Signed-off-by: Mike Lothian <mike@fireburn.co.uk>
---
 src/Common/precompiled.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/Common/precompiled.h b/src/Common/precompiled.h
index 790a001a8..d2d4631c7 100644
--- a/src/Common/precompiled.h
+++ b/src/Common/precompiled.h
@@ -31,6 +31,7 @@
 #endif
 
 // c includes
+#include <cstdarg>
 #include <cstdint>
 #include <cstdlib>
 #include <cmath>
@@ -584,4 +585,4 @@ namespace stdx
         constexpr std::underlying_type_t<EnumT> to_underlying(EnumT e) noexcept {
         return static_cast<std::underlying_type_t<EnumT>>(e);
     };
-}
\ No newline at end of file
+}
